pkg/processor/compute.go

```diff
@@ -17,19 +17,14 @@ import (
 	"github.com/go-kit/log"
 	"github.com/go-kit/log/level"
 
-	"time"
-
 	_ "github.com/mithrandie/csvq-driver"
 )
 
-func Compute(config *state.Config, registry *prometheus.Registry, logger log.Logger) error {
+func Compute(ctx context.Context, config *state.Config, registry *prometheus.Registry, logger log.Logger) error {
 	if err := updateSymlinks(config); err != nil {
 		return err
 	}
 
-	ctx, cancel := context.WithTimeout(context.Background(), 5 * time.Minute)
-	defer cancel()
```

Comment:
Could we keep this timeout and make it even more useful?

- We can now replace `context.Background()` with the new `ctx` argument
- Instead of hardcoded 5 minute timeout, I think it makes sense to use `interval` from config

This way if a computation takes more than the interval, it will be terminated with an error. In this case the operator should consider increasing `interval` then / look for a more powerful host.

This is also another safety measure to prevent stalled metrics due to unexpected issues in `csvq` (deadlock or something similar).

Reply:
yes, I was thinking about that too but hadn't considered that indeed erroring due to a timeout could actually be a benefit, not just a drawback to avoid by allowing for limitless computation time.